<launch>
  <arg name="id"              default="7" />
  <arg name="robot_type"      default="quad" />
  <arg name="ns"              value="/$(arg robot_type)$(arg id)" />
  <arg name="b_camera"        default="true" />
  <arg name="b_use_gt_bb"     default="false" />
  <arg name="b_enforce_0_yaw" default="true" />
  <arg name="detection_period" default="5" /> <!-- period to use to redo safety detection in seconds -->
  <arg name="b_rosbag"        default="false" />  <!-- boolean if we are reading data from a rosbag -->
  <arg name="rb_path"         default="/bags" />  <!-- path to the rosbag -->
  <arg name="rb_name"         default="quad7_2019-11-13-13-30-13" /> <!-- name of the rosbag -->
  <arg name="rb_start_delay"  default="2.0" /> <!-- how long to delay the start of the rosbag -->
  
  <group if="$(arg b_rosbag)">
    <param name="use_sim_time" value="true" />   <!--use the rosbag time instead of the computer's time. If using this use clock option in rosbag-->
  </group>
  <group unless="$(arg b_rosbag)">
    <param name="use_sim_time" value="false" />
  </group>
  
  <node pkg="tf" type="static_transform_publisher" name="trans_pub_map_world"
        args="0 0 0 0 0 0 /map /world 10"> 
  </node>

  <!-- IF using camera, launch our node -->
  <group if="$(arg b_camera)">
    <group ns="$(arg ns)">
      <include file="$(find ocam_msl)/launch/ocam_ros.launch" >
          <arg name="resolution"       value="0"/> <!-- These are the default cropping configurations from Ocam:  resolution: 0: 1280x960, 1: 1280x720, 2: 640x480, 3: 640x360    frame_rate: USB3.0 - 1280x960@45fps, 1280x720@60fps, 640x480@80fps, 320x240@160fps   USB2.0 - 1280x960@22.5fps, 1280x720@30fps, 640x480@80fps, 320x240@160fps -->
          <arg name="frame_rate"       value="22.5"/>
          <arg name="image_width"      value="640"/>
          <arg name="image_height"     value="480" />
          <arg name="camera_frame_id"  value="$(arg ns)/camera" />
      </include>
    </group>
  </group>

  <group if="$(arg b_rosbag)">
    <node pkg="rosbag" type="record" name="ocam_bag_rec" args="-o  $(find msl_raptor)/rosbag_for_post_process_ 
        /tf /tf_static 
        /quad7/camera/camera_info /quad7/camera/image_raw 
        /quad7/mavros/vision_pose/pose /quad7/mavros/local_position/pose /quad7/mavros/local_position/pose_cov 
        /quad4/mavros/vision_pose/pose /quad4/mavros/local_position/pose /quad4/mavros/local_position/pose_cov 
        /quad7/mavros/rc/in /quad7/mavros/rc/out /quad4/mavros/rc/in /quad4/mavros/rc/out 
        "/>
    </group>


</launch>
